code/modules/photography/camera/camera.dm

```diff
@@ -227,3 +227,69 @@
 		p.set_picture(picture, TRUE, TRUE)
 		if(CONFIG_GET(flag/picture_logging_camera))
 			picture.log_to_file()
+
+
+/obj/item/camera/syndicate/attack(mob/living/carbon/human/M, mob/user) // syndicate camera flash
```

Comment:
dont like that flash code is defined in two places now, make it so that there's a flash inside the camera and then call the flash's procs when it's used?

Reply:
I wanted to do that too but I don't remember how to do that (or if it is possible while being practical) I'll update it if someone tells me how

Reply:
keep a reference to a flash,
in intialize spawn one and forcemove it into the camera (probably want to add EMP proof or whatever if you need to)
in attack just call the flash' attack, etc.

Reply:
> keep a reference to a flash, in intialize spawn one and forcemove it into the camera (probably want to add EMP proof or whatever if you need to) in attack just call the flash' attack, etc.

I'll do it if I have to, but in all honestly, that sounds like more of a janky workaround than what I did

Reply:
@ToasterBiome I think if I add an actual invisible flash to the camera I'd introduce more problems down the line.. like people being able to somehow detect this is a syndicate item by somehow interacting with the flash item with another item. Even if I put it in the Camera's storage someone can maybe "dump out" the contents of the camera and the flash would be removed from the camera.

Reply:
> @ToasterBiome I think if I add an actual invisible flash to the camera I'd introduce more problems down the line.. like people being able to somehow detect this is a syndicate item by somehow interacting with the flash item with another item. Even if I put it in the Camera's storage someone can maybe "dump out" the contents of the camera and the flash would be removed from the camera.

I still don't like the idea of having flash code in two places

Reply:
> I think if I add an actual invisible flash to the camera I'd introduce more problems down the line.. like people being able to somehow detect this is a syndicate item by somehow interacting with the flash item with another item. Even if I put it in the Camera's storage someone can maybe "dump out" the contents of the camera and the flash would be removed from the camera.

I agree with ToasterBiome that it would be better if it held the flash inside of it, and some other items do this and have no issues from what I've seen, just make sure to delete it if the camera gets deleted